File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;
int cnt;
int n;
vector<int> numbers;

void run(int lv, int sum, vector<int> numbers, int target) {
    
    if(lv == n) {
        if(sum == target) {
            cnt ++;
        }
        return;
    }
    
    run(lv+1, sum+numbers[lv], numbers, target);
    run(lv+1, sum-numbers[lv], numbers, target);

}

int solution(vector<int> numbers, int target) {
    int answer = 0;
    
    n = numbers.size();
    
    run(0, 0, numbers, target);
    
    answer = cnt;

    return answer;
}

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
cnt = 0

def run(numbers, target, lv, res) :
    global cnt
    
    if(len(numbers)==lv) :
        if(res==target) :
            cnt += 1
            
        return
        
    run(numbers, target, lv+1, res + numbers[lv])
    run(numbers, target, lv+1, res - numbers[lv])
    
def solution(numbers, target):
    answer = 0
    run(numbers, target, 0, 0)
    
    answer = cnt
    
    return answer